core-patch/src/test/scala/eu/ostrzyciel/jelly/core/patch/PatchDecoderSpec.scala

```diff
@@ -0,0 +1,354 @@
+package eu.ostrzyciel.jelly.core.patch
+
+import eu.ostrzyciel.jelly.core.JellyExceptions.RdfProtoDeserializationError
+import eu.ostrzyciel.jelly.core.helpers.Mrl
+import eu.ostrzyciel.jelly.core.patch.handler.*
+import eu.ostrzyciel.jelly.core.patch.helpers.*
+import eu.ostrzyciel.jelly.core.proto.v1.*
+import eu.ostrzyciel.jelly.core.proto.v1.patch.*
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import scala.annotation.experimental
+
+@experimental
+class PatchDecoderSpec extends AnyWordSpec, Matchers:
+  val decoders: Seq[(
+    String,
+    (AnyPatchHandler[Mrl.Node], Option[RdfPatchOptions]) => PatchDecoder,
+    PatchStatementType
+  )] = Seq(
+    ("TriplesDecoder", MockPatchConverterFactory.triplesDecoder, PatchStatementType.TRIPLES),
+    ("QuadsDecoder", MockPatchConverterFactory.quadsDecoder, PatchStatementType.QUADS),
+    ("AnyDecoder", MockPatchConverterFactory.anyStatementDecoder, PatchStatementType.TRIPLES),
```

Comment:
why for any decoder patchstatementtype.triples

Reply:
Because it must decode something. Only triples and quads are valid values.